### PR TITLE
fix: Check API usage before restricting serving flags and admin

### PR DESCRIPTION
**Pull request by zachaysan**
## Changes

Added a check for the latest API usage since the 7 day grace period could have been quieter, reducing the usage to within limits.

See this issue for more context and information: https://github.com/Flagsmith/flagsmith-private/issues/14

## How did you test this code?

Altered an existing test.

### CODE REVIEW

File: api/organisations/tasks.py [hunk 1]
```diff
@@ -290,6 +290,17 @@ def restrict_use_due_to_api_limit_grace_period_over() -> None:
         if not stop_serving and not block_access:
             continue
 
+        if not organisation.has_subscription_information_cache():
+            continue
+
+        subscription_cache = organisation.subscription_information_cache
+        api_usage = get_current_api_usage(organisation.id, "30d")
```
Comment (khvn26) [inline api/organisations/tasks.py:297]: Do you think `"30d"` should be a constant?

File: api/organisations/tasks.py [hunk 2]
```diff
@@ -290,6 +290,17 @@ def restrict_use_due_to_api_limit_grace_period_over() -> None:
...
+        if api_usage / subscription_cache.allowed_30d_api_calls < 1.0:
+            logger.info(
+                f"API use for organisation {organisation.id} has fallen to below limit, so not restricting use."
+            )
```
Comment (khvn26) [inline api/organisations/tasks.py:301]: Do we _need_ to log that? I'd rather log the other branch?

**Review by khvn26 — APPROVED** [review]
LGTM overall with a couple of questions.

File: api/organisations/tasks.py (hunk 1 above)
Comment (zachaysan) [reply api/organisations/tasks.py:297]: In an upcoming branch the calls that say "30d" or similar will be replaced with datetimes so I think we can just leave it for now.

File: api/organisations/tasks.py (hunk 2 above)
Comment (zachaysan) [reply api/organisations/tasks.py:301]: This is the less frequently occurring case where we may need to debug when the task didn't select them, so I think it's a good candidate for inclusion.

**Pull request merged**